### PR TITLE
Kd Tree Nice Print

### PR DESCRIPTION
**Pull request by danston**
This little PR fixes some warnings in the benchmarks of the `Spatial_searching` package and adds a new print function that enables printing the produced tree in the [Graphviz](https://graphviz.org/) format. The results of the print can be found [here](https://github.com/CGAL/cgal/pull/5854). The changed docs are highlighted in red.

- [x] Remove tmp modifs in docs.
- [x] Update changes.md.

## Release Management

* Affected package(s): `Spatial_searching`
* Issue(s) solved (if any): no fixes
* Feature/Small Feature (if any):    [link](https://cgal.geometryfactory.com/CGAL/Members/wiki/Features/Small_Features/Kd_tree::write_graphviz()#Summary_of_API_changes) -- pre-approved Andreas Fabri 09:15, 10 December 2021 (CET) 
* Link to compiled documentation (obligatory for small feature): [docs](https://cgal.github.io/6043/v0/Spatial_searching/classCGAL_1_1Kd__tree.html#ac2851b5cafb8d5cce0dc5fb107c8f13f)
* License and copyright ownership: no changes

### CODE REVIEW

**Comment by afabri** [comment]
@danston can you remerge master so that the merge conflict goes away.
@sloriot should "graphviz"  be part of the function name instead of just 'print()`?

**Comment by sloriot** [comment]
boost uses [`write_graphviz()`](https://www.boost.org/doc/libs/1_77_0/libs/graph/doc/write-graphviz.html) so maybe we should use the same name.

File: Spatial_searching/include/CGAL/Kd_tree.h [hunk 1]
```diff
@@ -21,6 +21,9 @@
 #include <CGAL/basic.h>
 #include <CGAL/assertions.h>
 #include <vector>
+#include <string>
+#include <unordered_map>
+#include <fstream>
```
Comment (sloriot) [inline Spatial_searching/include/CGAL/Kd_tree.h:26]: ostream 
```suggestion
#include <ostream>
```

File: Spatial_searching/include/CGAL/Kd_tree_node.h [hunk 2]
```diff
@@ -15,7 +15,9 @@
 
 #include <CGAL/license/Spatial_searching.h>
 
-
+#include <string>
+#include <unordered_map>
+#include <fstream>
```
Comment (sloriot) [inline Spatial_searching/include/CGAL/Kd_tree_node.h:20]:
```suggestion
#include <ostream>
```

File: Spatial_searching/include/CGAL/Kd_tree_node.h [hunk 3]
```diff
@@ -15,7 +15,9 @@
 
 #include <CGAL/license/Spatial_searching.h>
 
-
+#include <string>
```
Comment (sloriot) [inline Spatial_searching/include/CGAL/Kd_tree_node.h:18]:
```suggestion
```

**Comment by afabri** [comment]
Sucessfully tested in https://cgal.geometryfactory.com/CGAL/testsuite/results-5.4-Ic-123.shtml

**Pull request merged**